components/brave_wallet/browser/internal/hd_key_zip32_unittest.cc

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/brave_wallet/browser/internal/hd_key_zip32.h"
+
+#include "brave/components/brave_wallet/common/brave_wallet.mojom.h"
+#include "brave/components/brave_wallet/common/zcash_utils.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+static_assert(BUILDFLAG(ENABLE_ORCHARD));
+
+namespace brave_wallet {
+
+TEST(HDKeyZip32Test, GenerateFromSeed) {
```

Comment:
is there any way we can make this less opaque? For instance if one of these tests failed I think it would be hard to debug. For instance is this based on bip32 where we could start from a key phrase in the tests and convert that to bytes? For they keys, is there any other confirmation we can do that wouldn't just be tautological? This is ok for follow-up

Reply:
maybe we should use the rust code directly to verify them to ensure we didn't create any problems in the ffi boundary?

Reply:
or maybe keep the test data as base64 and convert to binary in the test code so it's a bit easier to check using other tools?

Reply:
Test data is used in the same format as in the provided testvectors, so it could be compared more easely

Reply:
> maybe we should use the rust code directly to verify them to ensure we didn't create any problems in the ffi boundary?

We are using orchard crate which already has been tested on this vectors and derivation code in the zcash/rust is quite primitive, i believe top-level checks are more useful here, so we can check our bridge between c++ and rust.

Reply:
> As an example, is this based on bip32 where we could start from a key phrase in the tests and convert that to bytes?

Unfortunately not because of the way the test vectors have been designed here. In this case, they started from a [seed with bytes](https://github.com/zcash/zcash-test-vectors/blob/master/zcash_test_vectors/unified_address.py#L26) and that's why the [root_seed] (https://github.com/zcash/librustzcash/blob/zcash_address-0.3.2/components/zcash_address/src/kind/unified/address/test_vectors.rs#L29-L31) is set specifically to the way it is.

In other words, we are using a seed phrase here, we're just not starting with words. It's also worth pointing out that [ZIP-32](https://zips.z.cash/zip-0032) is different than [BIP-32](https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki)